### PR TITLE
[ESIMD] Revise log/exp implementation to be consistent with std/sycl

### PR DESCRIPTION
**Pull request by sndmitriev**
- Rename esimd::log/exp to esimd::log2/exp2 since they operate with base 2
- Add esimd::log/exp functions emulated via esimd::log2/exp2

Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 1]
```diff
@@ -1592,6 +1592,37 @@ asin(T src0, int flag = saturation_off) {
   return Result[0];
 }
 
+// log
+
+template <int SZ>
+ESIMD_NODEBUG ESIMD_INLINE simd<float, SZ> log(simd<float, SZ> src0,
+                                               int flag = saturation_off) {
+  constexpr float ln2 = 0.693147f; // std::numbers::ln2_v<float> in c++20
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1600]: Nit: is this precision maximum float can reach? If so, could you please add a comment?

File: sycl/test/esimd/math_impl.cpp [hunk 2]
```diff
@@ -25,7 +25,7 @@ SYCL_ESIMD_FUNCTION SYCL_EXTERNAL simd<float, 16> sycl_math(simd<float, 16> x) {
   return v;
 }
 
-// Math sin,cos,log,exp functions from esimd namespace are translated
+// Math sin,cos,log2,exp2 functions from esimd namespace are translated
```
Comment (kbobrovs) [inline sycl/test/esimd/math_impl.cpp:28]: please add E2E test link

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 3]
```diff
@@ -1592,6 +1592,37 @@ asin(T src0, int flag = saturation_off) {
   return Result[0];
 }
 
+// log
+
+template <int SZ>
+ESIMD_NODEBUG ESIMD_INLINE simd<float, SZ> log(simd<float, SZ> src0,
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1598]: please add a doc comment that this is s/w-emulated version based on h/w supported log2

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp (hunk 1 above)
Comment (sndmitriev) [reply sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1600]: For ln2 looks like it should be possible to add more digits after the dot. But for log2e (below) it seems to be the max precision already.

```
$ cat test.cpp 
#include <stdio.h>
int main() {
  float a1 = 0.693147180559945309417232121458176f;
  float b1 = 0.69314718f;
  int *ai1 = (int *)&a1;
  int *bi1 = (int *)&b1;
  printf("*ai1 == *bi1 %d\n", (*ai1 == *bi1));
  float a2 = 1.442695040888963407359924681001892f;
  float b2 = 1.442695f;
  int *ai2 = (int *)&a2;
  int *bi2 = (int *)&b2;
  printf("*ai2 == *bi2 %d\n", (*ai2 == *bi2));
  return 0;
}
$ g++ test.cpp; ./a.out 
*ai1 == *bi1 1
*ai2 == *bi2 1
$ 
```

I can actually add a double initializer and let compiler truncate it to float.

**Comment by sndmitriev** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/673

File: sycl/test/esimd/math_impl.cpp (hunk 2 above)
Comment (sndmitriev) [reply sycl/test/esimd/math_impl.cpp:28]: https://github.com/intel/llvm-test-suite/pull/673

**Review by kbobrovs — APPROVED** [review]

**Comment by bader** [comment]
@vladimirlaz, could you merge this PR, please?

**Pull request merged**